var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -833,5 +857,20 @@ def setup_build_environment(self, env):
     def setup_run_environment(self, env):
         env.set("LAMMPS_POTENTIALS", self.prefix.share.lammps.potentials)
         if "+python" in self.spec:
-            env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib)
-            env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib64)
+            if self.spec.platform == "darwin":
+                env.prepend_path("DYLD_LIBRARY_PATH", self.prefix.lib)
+                env.prepend_path("DYLD_LIBRARY_PATH", self.prefix.lib64)
```

Comment:
```suggestion
                env.prepend_path("DYLD_FALLBACK_LIBRARY_PATH", self.prefix.lib)
                env.prepend_path("DYLD_FALLBACK_LIBRARY_PATH", self.prefix.lib64)
```
This is only used if the libs can't be found by normal means.